src/dmd/inline.d

```diff
@@ -2214,10 +2214,9 @@ private bool argNeedsDtor(Expression arg)
              * perhaps they can be combined.
              */
 
-            int symbolDg(Dsymbol s)
+            void symbolDg(Dsymbol s)
             {
                 Dsymbol_needsDtor(s);
-                return 0;
             }
```

Comment:
You could make the call sites easier to understand if you replace the call to this function with `(s) { Dsymbol_needsDtor(s); }`, or even `s => Dsymbol_needsDtor(s)`.

Reply:
I did it this way so there weren't two functions generated.

Reply:
Actually, I think you  could do pass simply `&Dsymbol_needsDtor` - that way no extra functions would be needed.